src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -105,6 +105,26 @@ internal abstract partial class AbstractReferenceFinder : IReferenceFinder
             return documents.ToImmutable();
         }
 
+        protected static async Task<ImmutableArray<Document>> FindAllSolutionDocumentsAsync<T>(
+            Solution solution,
+            Func<Document, T, CancellationToken, ValueTask<bool>> predicateAsync,
+            T value,
+            CancellationToken cancellationToken)
+        {
+            using var _ = ArrayBuilder<Document>.GetInstance(out var documents);
+
+            foreach (var project in solution.Projects)
+            {
+                foreach (var document in await project.GetAllRegularAndSourceGeneratedDocumentsAsync(cancellationToken).ConfigureAwait(false))
+                {
+                    if (await predicateAsync(document, value, cancellationToken).ConfigureAwait(false))
+                        documents.Add(document);
+                }
+            }
+
+            return documents.ToImmutable();
+        }
```

Comment:
why is this helper here?

Reply:
The current FAR implementation for pp symbols searches through the entire solution's documents, and this helper is not specifically bound to pp symbols, so it was placed there abstractly

Reply:
> and this helper is not specifically bound to pp symbols

What other symbols use this helper?

Reply:
No others use it obviously, but I mean the logic doesn't scream "pp symbols" anywhere, so the saner option to me was to place it abstractly on the base class, instead of privatizing it on the only finder making use of it

Reply:
Our style is to do the reverse.  Keep things as close to the code that actually needs it.  Only if it is needed by other things, move into a more wide-scoped location.